lib/rust/flexer/src/automata/dfa.rs

```diff
@@ -64,62 +64,89 @@ impl From<Vec<Vec<usize>>> for Matrix<state::Id> {
 
 
 
-// ===========
-// == Tests ==
-// ===========
+// ================
+// === Callback ===
+// ================
+
+/// The callback associated with an arbitrary state of a finite automaton.
+///
+/// It contains the rust code that is intended to be executed after encountering a
+/// [`pattern`](super::pattern::Pattern) that causes the associated state transition. This pattern
+/// is declared in [`Rule.pattern`](crate::group::rule::Rule::pattern).
+#[derive(Clone,Debug,PartialEq,Eq)]
+pub struct RuleExecutable {
+    /// A description of the priority with which the callback is constructed during codegen.
+    pub priority: usize,
+    /// The rust code that will be executed when running this callback.
+    pub name: String,
+}
+
+
+
+// =============
+// === Tests ===
+// =============
 
 #[cfg(test)]
 pub mod tests {
     use super::*;
     use crate::automata::state;
 
-    const I:usize = state::INVALID.id;
+    const I:usize = state::Identifier::INVALID.id;
 
     /// DFA automata that accepts newline '\n'.
     pub fn newline() -> DFA {
         DFA {
-            alphabet: Alphabet::from(vec![10,11]),
+            alphabet_segmentation: alphabet::Segmentation::from_divisions(
+                vec![10,11].as_slice()
+            ),
             links: Matrix::from(vec![vec![I,1,I], vec![I,I,I]]),
             callbacks: vec![
                 None,
-                Some(Callback{priority:2,name:"group0_rule0".into()}),
+                Some(RuleExecutable {priority:2,name:"group0_rule0".into()}),
```

Comment:
nice name. Btw, I'm not asking to change it because this is not your code, but all such things should be just 
`RuleExecutable::new(2,"group0_rule0")` instead. If you'd like to have stronger typing, then `RuleExecutable::new(Priority(2),"group0_rule0")` or so,mething similar.